packages/react-shortcuts/src/ShortcutManager/ShortcutManager.tsx

```diff
@@ -0,0 +1,124 @@
+import {Key} from '../types';
+
+export interface Data {
+  node: HTMLElement | null | undefined;
+  keys: Key[];
+  ignoreInput: boolean;
+  onMatch(): void;
+  allowDefault: boolean;
+}
+
+export default class ShortcutManager {
+  private keysPressed: Key[] = [];
+  private shortcuts: Data[] = [];
+  private shortcutsMatched: Data[] = [];
+  private timer!: number;
+
+  setup() {
+    document.addEventListener('keydown', this.handleKeyDown);
+  }
+
+  subscribe(data: Data) {
+    this.shortcuts.push(data);
+    const self = this;
```

Comment:
I'm not a fan of this pattern, although it is quite common. I'd rather see us desructure `this.shortcuts` above and rely on that being in the closure, rather than suing the `self` keyword.

Reply:
Me neither, looked over it while porting.